### PR TITLE
feat: add github package

### PR DESCRIPTION
**Pull request by verbanicm**

### CODE REVIEW

File: pkg/github/github.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/sethvargo/go-retry"
+	"golang.org/x/oauth2"
+)
+
+var ignoredStatusCodes = []int{400, 401, 403, 404, 422}
```
Comment (sethvargo) [inline pkg/github/github.go:28]: I'd suggest making this a `map[int]struct{}`

File: pkg/github/github.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	for _, opt := range opts {
+		if opt != nil {
+			cfg = opt(cfg)
+		}
+	}
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
```
Comment (sethvargo) [inline pkg/github/github.go:69]: Is there an SDK function for "source the creds from the env"? That would let us support private key auth and token auth easily.

File: pkg/github/github.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	g := &GitHubClient{
+		cfg:    cfg,
+		client: client,
+	}
+
+	return g
+}
+
+// CreatePullRequestComment creates a pull request comment.
+func (g *GitHubClient) CreatePullRequestComment(ctx context.Context, owner, repo string, number int, body string) (int64, error) {
```
Comment (sethvargo) [inline pkg/github/github.go:82]: It feels weird that this only returns an ID?

File: pkg/github/github.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+	return nil
+}
+
+// ListPullRequestComments lists existing comments for a pull request.
+func (g *GitHubClient) ListPullRequestComments(ctx context.Context, owner, repo string, number int, opts *github.IssueListCommentsOptions) ([]*github.IssueComment, *github.Response, error) {
+	backoff := retry.NewFibonacci(g.cfg.initialRetryDelay)
+	backoff = retry.WithMaxRetries(g.cfg.maxRetries, backoff)
+	backoff = retry.WithCappedDuration(g.cfg.maxRetryDelay, backoff)
+
+	commentsResponse := make([]*github.IssueComment, 0)
```
Comment (sethvargo) [inline pkg/github/github.go:164]: Nit: 
```suggestion
	var commentsResponse []*github.IssueComment
```

**Review by bradegler — APPROVED** [review]

File: pkg/github/github.go (hunk 2 above)
Comment (verbanicm) [reply pkg/github/github.go:69]: Do you mean so we could auth as a user or as an app? It seems there are two different methods for this: https://github.com/google/go-github#authentication

Comment (sethvargo) [reply pkg/github/github.go:69]: Yea, I'm thinking we might auth as an app in the future.

Comment (verbanicm) [reply pkg/github/github.go:69]: We have: https://github.com/abcxyz/pkg/blob/main/githubapp/githubapp.go

We can wrap this, but I think we'll need to accept an HTTP Client and build a wrapper based on inputs to generate the HTTP client we want to use...doesnt seem like it exists.

Ill add an issue for this.

Comment (verbanicm) [reply pkg/github/github.go:83]: Created #130 and added comment inline.

**Review by sethvargo — APPROVED** [review]

**Pull request merged**